discovery/syncer.go

```diff
@@ -507,7 +510,9 @@ func (g *GossipSyncer) channelGraphSyncer() {
 				// for the new channels.
 				queryReply, ok := msg.(*lnwire.ReplyChannelRange)
 				if ok {
-					err := g.processChanRangeReply(queryReply)
+					err := g.processChanRangeReply(
+						ctx, queryReply,
```

Comment:
similar question here - we'd exit when `case <-g.cg.Done():`, should we also exit when the context is canceled or done?

Reply:
so  for internal methods like this, i dont think we need to listen on both because the context being passed in to these internal methods are always derived from the parent and we store the `cancel` - and so if `cg` is "done" then cancel will also be called at the same time (since we do these 2 together in `Stop`). It's only when a ctx is being passed in from an external subsystem via an exported method where we should listen on both since then they have different triggers.

Reply:
one could argue that it could be possible for the parent context to be closed separately from Stop being called (and ie, then cg.Done() and ctx.Done() would fire separately) but the parent context _should_ always be the main single LND context in `Main` and so if that gets cancelled, LND's Stop method should be triggered and all the Stop methods of each subsystem will get called in which case the cg.Done() will fire in any case

Reply:
Cool, I was confused by the external ctx from the caller vs the internal ctx derived from the `Main`. And I think eventually this ctx can be passed and controlled by a remote service correct?

Reply:
yeah correct :)